src/Common/Core/Impl/Services/ServiceManager.cs

```diff
@@ -40,24 +33,36 @@ public class ServiceManager : IServiceManager {
             type = type ?? typeof(T);
             Check.ArgumentNull(nameof(service), service);
             Check.InvalidOperation(() => _s.TryAdd(type, service), "Service already exists");
-            ServiceAdded?.Invoke(this, new ServiceContainerEventArgs(type));
+
+            var mainThread = service as IMainThread;
```

Comment:
That's not right. Fixtures are disposed inside test pipeline, so if test isn't fully completed at that time, we may get incorrect results: https://github.com/Microsoft/RTVS/issues/3523